src/poetry/console/commands/env/activate.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import shlex
+
+from typing import TYPE_CHECKING
+
+import shellingham
+
+from poetry.console.commands.command import Command
+from poetry.utils._compat import WINDOWS
+
+
+if TYPE_CHECKING:
+    from poetry.utils.env import Env
+
+
+class ShellNotSupportedError(Exception):
+    """Raised when a shell doesn't have an activator in virtual environment"""
+
+
+class EnvActivateCommand(Command):
+    name = "env activate"
+    description = "Print the command to activate a virtual environment"
+
+    def handle(self) -> int:
+        from poetry.utils.env import EnvManager
+
+        env = EnvManager(self.poetry).get()
+
+        if command := self.get_activate_command(env):
+            self.line(command)
+            return 0
+        else:
+            raise ShellNotSupportedError(
+                "Discovered shell doesn't have an activator in virtual environment"
+            )
+
+    def get_activate_command(self, env: Env) -> str:
```

Comment:
```suggestion
    def _get_activate_command(self, env: Env) -> str:
```

I'd add an underscore to make clear it is internal.